### PR TITLE
Scripting api: error extraction from QueryResult

### PR DESCRIPTION
**Pull request by todvora**
## Description
Extracting errors from search results

## Motivation and Context
Provide better error messages to users

## How Has This Been Tested?
Added integration test

## Screenshots (if appropriate):
image

/nocl

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Refactoring (non-breaking change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/SearchJobToTabularResponseMapper.java [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+import org.slf4j.LoggerFactory;
+
+import javax.ws.rs.BadRequestException;
+import javax.ws.rs.NotFoundException;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class SearchJobToTabularResponseMapper {
```
Comment (luk-kaminski) [inline graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/SearchJobToTabularResponseMapper.java:42]: 1. It does not need to happen in this PR, but we may want to unit test this class.
2. I am not 100% sure if we want to to use "javax.ws.rs.*" exceptions in this mapper, or should it rather be unaware/independent of this package, throwing more general exceptions, that will be mapped to proper exceptions by the resource class - just to consider.

**Review by luk-kaminski — APPROVED** [review]
Great job, thanks!
I've left some comments, but nothing they may be even considered even after merging this to master, as further improvements.

File: graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/SearchJobToTabularResponseMapper.java (hunk 1 above)
Comment (todvora) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/SearchJobToTabularResponseMapper.java:42]: yeah, custom exceptions and their mappers would be the cleanest solution. Let me do that here when we are already refactoring error handling.

Comment (todvora) [reply graylog2-server/src/main/java/org/graylog/plugins/views/search/rest/scriptingapi/mapping/SearchJobToTabularResponseMapper.java:40]: added custom exception that's re-thrown in the resource. Primitive but cleaner than before.

**Pull request merged**